### PR TITLE
Fixed mean image support for IR v10

### PR DESCRIPTION
**Pull request by ilyachur**
Task: 37278

### CODE REVIEW

**Comment by ilyachur** [comment]
@GlebKazantaev  Please have a look

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp [hunk 1]
```diff
@@ -196,7 +196,135 @@ std::shared_ptr<ICNNNetwork> V10Parser::parse(const pugi::xml_node& root, std::i
...
+            }
+        }
+        if (!preProcessInput) {
+            preProcessInput = inputs.begin()->second;
+        }
+
+        inputName = preProcessInput->name();
+    } else {
+        preProcessInput = network.getInputsInfo().at(inputName);
+        if (!preProcessInput)
```
Comment (GlebKazantaev) [inline inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:242]: std::map will throw an out_of_range exception if key doesn't exists. So this statement won't work.

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp [hunk 2]
```diff
@@ -196,7 +196,135 @@ std::shared_ptr<ICNNNetwork> V10Parser::parse(const pugi::xml_node& root, std::i
...
+        }
+    }
+
+    if (idsForMeanImage.size() == noOfChannels) {
+        pp.setVariant(MEAN_IMAGE);
+    } else if (idsForMeanImage.size() == 0) {
+        pp.setVariant(NONE);
+    } else {
+        std::string validMeanImageIds = "";
+        for (auto id : idsForMeanImage) {
```
Comment (GlebKazantaev) [inline inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:321]: Why do we need cycle if we throw an exception later?

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:242]: Agree, fixed

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp (hunk 2 above)
Comment (ilyachur) [reply inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:321]: We generate the right message for an exception in this cycle.

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**